src/webgpu/api/validation/capability_checks/limits/maxBindGroups.spec.ts

```diff
@@ -0,0 +1,33 @@
+import { range } from '../../../../../common/util/util.js';
+
+import { kLimitBaseParams, makeLimitTestGroup } from './limit_utils.js';
+
+const limit = 'maxBindGroups';
+export const { g, description } = makeLimitTestGroup(limit);
+
+g.test('createBindGroupLayout,at_over')
+  .desc(`Test using at and over ${limit} limit`)
+  .params(kLimitBaseParams)
+  .fn(async t => {
+    const { limitTest, testValueName } = t.params;
+    await t.testDeviceWithRequestedLimits(
+      limitTest,
+      testValueName,
+      async ({ device, testValue, shouldError }) => {
+        await t.expectValidationError(() => {
+          const bindGroupLayouts = range(testValue, (i: number) =>
+            device.createBindGroupLayout({
+              entries: [
+                {
+                  binding: 0,
+                  visibility: GPUShaderStage.VERTEX,
+                  buffer: {},
+                },
+              ],
+            })
+          );
+          device.createPipelineLayout({ bindGroupLayouts });
+        }, shouldError);
+      }
+    );
+  });
```

Comment:
Occurs to me create\*Pipeline\* with 'auto' layout should also be able to hit this limit - if you declare and use a `@group` index that's too large, somewhere along the line it should hit this limit. I'm not sure if the spec actually says this though.

I also am not sure if the error should/would be in create\*Pipeline\* (for binding declarations used by the entry point) or createShaderModule (for any binding declaration).

Reply:
so createShaderModule is failing here so I added a test for that. I kind of feel like createShaderModule should not be failing though

https://github.com/gpuweb/gpuweb/issues/3815

Reply:
actually createShaderModule is not enforcing these limits. In anycase, I'll add the test in the next push